src/types.ts

```diff
@@ -80,8 +85,10 @@ export interface Command {
 }
 
 export interface SlashCommand extends Command {
-    options?: CommandOption[]
+    options?: CommandOption[],
     execute: (interaction: ChatInputCommandInteraction) => Promise<void>,
+    guild?: boolean,
+    role?: boolean
```

Comment:
Naming seems weird. Should either be `requiresRole`, or just stick with the classic `privileged` (if we go with the role-optional approach). It would conflict with the original `privileged`, though so we'd have to rename the other one